File: java backend/src/main/java/com/fleman/controller/EmailController.java
/**
 * 
 */
package com.fleman.controller;

/**
 * 
 */

import org.springframework.beans.factory.annotation.Autowired;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import com.fleman.service.impl.EmailService;


@RestController
public class EmailController {

    @Autowired
    private EmailService emailService;

    @GetMapping("/api/email/test")
   
    public String sendTestEmail() {

        emailService.sendEmail(
                "teamgryffindor45@gmail.com",   //  email
                "Test Email from WanderCar",
                "Congratulations! Your Email Service is working successfully."
        );

        return "Email sent successfully!";
    }
}

File: WanderCar_DotNet/src/FlemanApi/Controllers/EmailController.cs
using FlemanApi.Service;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace FlemanApi.Controllers;

// Debug/test endpoint — mirrors com.fleman.controller.EmailController.
// Not in SecurityConfig's permitAll list, so it requires authentication
// (no specific role), same as the Java app's default anyRequest().authenticated().
[ApiController]
[Authorize]
public class EmailController : ControllerBase
{
    private readonly IEmailService _emailService;

    public EmailController(IEmailService emailService)
    {
        _emailService = emailService;
    }

    [HttpGet("api/email/test")]
    public async Task<IActionResult> SendTestEmail()
    {
        await _emailService.SendEmailAsync(
            "teamgryffindor45@gmail.com",
            "Test Email from WanderCar",
            "Congratulations! Your Email Service is working successfully.");

        return Ok("Email sent successfully!");
    }
}
